src/flip_edge.cpp

```diff
@@ -0,0 +1,71 @@
+#include <npe.h>
+#include <common.h>
+#include <typedefs.h>
+
+#include <pybind11/stl.h>
+
+#include <igl/flip_edge.h>
+
+const char* ds_flip_edge = R"igl_Qu8mg5v7(
+
+Flip an edge in a triangle mesh.  The edge specified by uei must have
+exactly two adjacent faces.  Violation will result in exception.
+Another warning: edge flipping could convert manifold mesh into
+non-manifold.
+
+Parameters
+----------
+F         #F by 3 list of triangles.
+E         #F*3 by 2 list of all of directed edges
+uE        #uE by 2 list of unique undirected edges
+EMAP      #F*3 list of indices into uE, mapping each directed edge to unique
+          undirected edge
+uE2E      #uE list of lists of indices into E of coexisting edges
+ue        #index into uE the edge to be flipped.
+
+Returns
+-------
+
+Updated   F, E, uE, EMAP and uE2E.
+
+
+See also
+--------
+
+
+Notes
+-----
+None
+
+Examples
+--------
+
+
+)igl_Qu8mg5v7";
+
+npe_function(flip_edge)
+npe_doc(ds_flip_edge)
+
+npe_arg(f, dense_long)
```

Comment:
why not `dense_int, dense_long, dense_longlong`?

Reply:
`std::vector<std::vector<uE2EType> > uE2E` and `F` have to be the same type, and since there is no equivalent for `npe_arg(ue2e, std::vector<std::vector<type F> >)`, or at least I couldn't figure it out, `long` for both of them. Is there a way to define uE2E with template type in arguments?

Reply:
it is better to copy `std::vector<std::vector<uE2EType> >` from long to `npe_Scalar_f` to keep to python side clean.

Also use `typename EigenDenseInt::Scalar` instead of long, it might be different in different system

Reply:
changed long to `typename EigenDenseInt::Scalar`, consequently `dense_long` to `dense_longlong`  and `npe_Scalar_f` was already used for the copy.